### ISSUE TITLE
window.focus() not working in sandbox mode

### ISSUE THREAD

**Issue by rawatsandeep670** [electron/electron#8969]
* Electron version: 1.6.1
* Operating system: window 7 (64 bit)

### Expected behavior
when i call window.focus() window getting focused.

### Actual behavior
window not come to focus while focus event trigger.

window not come in focus.
### How to reproduce
**main.js**
mainWindow = new BrowserWindow({width: 100, height: 100, webPreferences: {nodeIntegration: true, sandbox: true}});
mainWindow.loadURL(`http://localhost:9000`);

**index.html**
<button onclick="focusWin()">focus</button>

**app.js**
var win = window.open("child.html", '', 'width=100,height=100,left=0;top=0');
function focusWin(){
   setTimeout(function(){
        win.focus();
   }, 100);
}

**Comment by hovancik** [comment]
Read this: https://github.com/electron/electron/blob/master/docs/api/browser-window.md#using-ready-to-show-event

**Comment by rawatsandeep670** [comment]
thanks @hovancik, but I use window.open(...) not new BrowserWindow(...)

**Comment by hovancik** [comment]
@rawatsandeep670 
What is window? It's not clear from your code. 

I think you should be able to use `ready-to-show` or `did-finish-load`

**Comment by rawatsandeep670** [comment]
yes

**Comment by hovancik** [comment]
maybe use this then? https://github.com/electron/electron/blob/master/docs/api/browser-window.md#parent-and-child-windows

not getting the idea behind using `window.open`

**Comment by rawatsandeep670** [comment]
ok, can you tell me one thing?

how to send an constructor in other window using BrowserWindow

**Comment by hovancik** [comment]
You should use https://github.com/electron/electron/blob/master/docs/api/ipc-main.md to communicate between windows. This way you can send data from parent to child or other way around.

**Comment by rawatsandeep670** [comment]
but how could I send an constructor object one to other window? when I try to send constructor object, IPC do auto serilized this object.

**Comment by hovancik** [comment]
what exactly are you trying to do?

**Comment by rawatsandeep670** [comment]
I want to share a constructor with child window, but IPC, serializes the constructor.

**Comment by hovancik** [comment]
No, I mean what are you "trying" to do. The point being: you are doing it not electron way, so you need to change that. 

Are you trying to show specific data on child window? You can send them via IPC.
Are you trying to calculate something? You should do that in main and send result via IPC. 

And so on...

**Comment by rawatsandeep670** [comment]
I am not send a JSON object or string. I want to send a constructor e.g.:

**function ElectronFunc(){
   this.addEventListener = function(){
   }
   this.name= "100100";
}

var ef = new ElectronFunc();**

when I send **ef** to another window using IPC, IPC delete all methods of constructor (like addEventListener); because IPC serializes data before send.

**Comment by hovancik** [comment]
Why are you not using js code per BrowserWindow? You can create event listener in any window. Why are you trying to send it there? It doesn't make sense.

**Comment by rawatsandeep670** [comment]
It make sense because if I make another Instance of this Function older instance auto distroye so I create it only once and want to share other window.

**Comment by hovancik** [comment]
I think you need to provide better code example, I am not able to undertsand what are you trying to accomplish. Maybe other people will understand better.

**Comment by rawatsandeep670** [comment]
ok thanks @hovancik

**Comment by rawatsandeep670** [comment]
@tarruda Hi, I was following your [comment](https://github.com/electron/electron/issues/1865#issuecomment-249989894) about using sandox flag to access complete window.opener and had a query regarding the same..

window.focus() not working please help me out

**Comment by tarruda** [comment]
@rawatsandeep670 sandbox is still an experimental option, so there may be a lot of native APIs that don't work. For now I suggest that you use the preload script to replace `window.focus` by a function that calls `ipcRenderer` to delegate focusing to the main process.

**Comment by rawatsandeep670** [comment]
thanks @tarruda, can u tell me one thing.

how I send a constructor one BrowserWindow to another BrowserWindow?

**Comment by codebytere** [comment]
We are no longer implementing bugfixes for versions of Electron <= `1.7.x`, so i'm going to close this issue but if it is still persisting in more recent versions of Electron we can certainly reopen it!

**Comment by nornagon** [comment]
I'm pretty sure this is still true in 3.0.x

**Comment by sofianguy** [comment]
Thank you for taking the time to report this issue and helping to make Electron better.

The version of Electron you reported this on has been superseded by newer releases. 

If you're still experiencing this issue in Electron v4.2.x or later, please add a comment specifying the version you're testing with and any other new information that a maintainer trying to reproduce the issue should know.

I'm setting the `blocked/need-info` label for the above reasons. This issue will be closed 7 days from now if there is no response.

Thanks in advance! Your help is appreciated.

**Comment by sofianguy** [comment]
Thank you for your issue!

We haven't gotten a response to our questions in our comment above. With only the information that is currently in the issue, we don't have enough information to take action. I'm going to close this but don't hesitate to reach out if you have or find the answers we need, we'll be happy to reopen the issue.
